Source/WTF/wtf/RetainPtr.h

```diff
@@ -65,9 +65,30 @@ typedef struct objc_object *id;
 // both ways, we need a separate copy of our code when ARC is enabled.
 #if __has_feature(objc_arc)
 #define adoptNS adoptNSArc
+#define retainFoundationPtr retainFoundationPtrArc
+#define releaseFoundationPtr releaseFoundationPtrArc
+#define autoreleaseFoundationPtr autoreleaseFoundationPtrArc
 #define RetainPtr RetainPtrArc
 #endif
 
+inline void retainFoundationPtr(CFTypeRef ptr) { CFRetain(ptr); }
+inline void releaseFoundationPtr(CFTypeRef ptr) { CFRelease(ptr); }
+#if HAVE(CFAUTORELEASE)
+inline void autoreleaseFoundationPtr(CFTypeRef ptr) { CFAutorelease(ptr); }
+#endif
+
+#ifdef __OBJC__
+#if __has_feature(objc_arc)
+inline void retainFoundationPtr(id) { }
+inline void releaseFoundationPtr(id) { }
+inline void autoreleaseFoundationPtr(id) { }
+#else
+inline void retainFoundationPtr(id ptr) { [ptr retain]; }
+inline void releaseFoundationPtr(id ptr) { [ptr release]; }
+inline void autoreleaseFoundationPtr(id ptr) { [ptr autorelease]; }
+#endif
+#endif
```

Comment:
Would be nice if these were private `static` methods on `RetainPtr<>` since it would be unexpected to use them outside of the class.